topic-operator/src/main/java/io/strimzi/operator/topic/Main.java

```diff
@@ -47,7 +53,9 @@ private void deploy(Config config) {
                         .setJvmMetricsEnabled(true)
                         .setEnabled(true));
         Vertx vertx = Vertx.vertx(options);
-        Session session = new Session(kubeClient, config);
+
+        WorkerExecutor executor = vertx.createSharedWorkerExecutor(blockingWorkerPoolName, blockingWorkerPoolSize);
```

Comment:
Why we don't use the same approach of ZkImpl? I think that having a dedicated executor per verticle/class is safer, as you can tune its configuration without the risk of messing up other classes. Bonus: you don't need to change tests.

Reply:
@fvaleri I was trying to make the executor a dependency I could replace at test time, but there was not actual point in doing so now that I've written some tests :)